Update docs about LaTeX parser (#25733)
sympy/sympy: doc/src/modules/parsing.rst

Experimental `\mathrm{\LaTeX}` Parsing
--------------------------------------

The current implementations are experimental. The behavior, parser backend(s) and
API might change in the future. Unlike some of the other parsers, `\mathrm{\LaTeX}` is
designed as a *type-setting* language, not a *computer algebra system* and so
can contain typographical conventions that might be interpreted multiple ways.

`\mathrm{\LaTeX}` Parsing Functions Reference
---------------------------------------------

.. autofunction:: sympy.parsing.latex.parse_latex

ANTLR Backend
^^^^^^^^^^^^^

The ANTLR-based `\mathrm{\LaTeX}` parser was ported from
`latex2sympy <https://github.com/augustt198/latex2sympy>`_. While functional
and its API should remain stable, the parsing behavior or backend may change in
future releases.

.. _ANTLR parser caveats:

ANTLR `\mathrm{\LaTeX}` Parser Caveats
""""""""""""""""""""""""""""""""""""""

> In its current definition, the parser will at times will fail to fully parse
> the expression, but not throw a warning::

Reviewer: ```suggestion
In its current definition, the parser may fail to fully parse
an expression, yet not throw a warning::
```

point out these ambiguities and notify the user, or even return all possible
interpretations.

The Lark-based parser exposes a number of internals which allow the user to customize
the parser's behavior. For example, the user can specify their own `\mathrm{\LaTeX}`
grammar by passing the path to the grammar file to the ``LarkLaTeXParser`` while
instantiating the parser.

The user can also specify their own custom transformer class to the `LarkLaTeXParser`
class.

The two examples mentioned above can be found in the
`test_custom_latex.py <https://github.com/sympy/sympy/blob/395e820b114d2b169483354f1f4ee2f439faa292/sympy/parsing/tests/test_custom_latex.py>`_
file.

Lark `\mathrm{\LaTeX}` Parser Capabilities
""""""""""""""""""""""""""""""""""""""""""

In order to use the Lark-based LaTeX parser, it is important to know what it can
and cannot do. As the parser is still experimental, it supports many things, but
some features are still only partially implemented, or not available.

As such, we will list the types of expressions that it can parse, and then list
some expression types of interest where it may fail.

Here is a list of the things which are supported:

> * Symbols which consist of one letter. I.e., things like ``a``, ``b``, ``x``, etc.

Reviewer: ```suggestion
* Symbols which consist of one letter, e.g., ``a``, ``b``, ``x``, etc.
```
